### DISCUSSION TITLE
[Question]: Configure OpenAI + vLLM (OpenAI endpoint) at the same time

### DISCUSSION THREAD

**Discussion by Tostino** [danny-avila/LibreChat#1432] [Q&A] [answered]
### What is your question?

I was wondering how I should configure things if I want access to both OpenAI endpoints, as well as models that I have hosted using vLLM with their OpenAI endpoint at the same time?

### More Details

This isn't specifically related to vLLM, but really any OpenAI compatible endpoint that you can self-host.

### What is the main subject of your question?

Endpoints

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment, marked as answer]
The only way to use both simultaneously is to have a dedicated endpoint for reverse proxy: https://github.com/danny-avila/LibreChat/issues/1344

Not yet implemented, will work on it soon.

**Reply by danny-avila** [reply]
I forgot to mention you could potentially configure both through LiteLLM: https://litellm.ai/

**Reply by Tostino** [reply]
Appreciate the info there!

Is it as simple as copying the OpenAIClient.js file and adding it to the index.js?  If so, I can attempt and submit a PR.

**Reply by danny-avila** [reply]
Appreciate that! No not that simple but not the hardest. This is a priority item for me but I don’t have my computer until the weekend. Will get right on it
